antiphishing/antiphishing.go

```diff
@@ -0,0 +1,183 @@
+package antiphishing
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/botlabs-gg/yagpdb/common"
+	"github.com/mediocregopher/radix/v3"
+	"github.com/sirupsen/logrus"
+)
+
+type Plugin struct {
+	stopWorkers chan *sync.WaitGroup
+}
+
+type BitFlowAntiFishResponse struct {
+	Match   bool `json:"match"`
+	Matches []struct {
+		Followed    bool    `json:"followed"`
+		Domain      string  `json:"domain"`
+		URL         string  `json:"url"`
+		Source      string  `json:"source"`
+		Type        string  `json:"type"`
+		TrustRating float64 `json:"trust_rating"`
+	} `json:"matches"`
+}
+
+var (
+	hyperphishURL      = "https://api.hyperphish.com/gimme-domains"
+	bitflowAntiFishURL = "https://anti-fish.bitflow.dev/check"
+)
+
+var logger = common.GetPluginLogger(&Plugin{})
+
+func RegisterPlugin() {
+	common.RegisterPlugin(&Plugin{
+		stopWorkers: make(chan *sync.WaitGroup),
+	})
+}
+
+func (p *Plugin) PluginInfo() *common.PluginInfo {
+	return &common.PluginInfo{
+		Name:     "AntiPhishing",
+		SysName:  "anti_phishing",
+		Category: common.PluginCategoryModeration,
+	}
+}
+
+func fetchHyperfishDomains() ([]string, error) {
+	resp, err := http.Get(hyperphishURL)
+	if err != nil {
+		return nil, err
+	}
+	domains := make([]string, 0)
+	err = json.NewDecoder(resp.Body).Decode(&domains)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	return domains, nil
+}
+
+func saveHyperfishDomains() ([]string, error) {
+	domains, err := fetchHyperfishDomains()
+	if err != nil {
+		return nil, err
+	}
+
+	//clear old domains incase there was a false positive
+	redisDelErr := common.RedisPool.Do(radix.Cmd(nil, "DEL", "hyperfish_domains"))
```

Comment:
Maybe make `hyperfish_domains` a const, e.g., `RedisKeyHyperfishDomains`, to be consistent with existing code and reduce magic values.

Also:
```suggestion
	err = common.RedisPool.Do(radix.Cmd(nil, "DEL", "hyperfish_domains"))
```
Typically in Go the idiomatic pattern is to shadow the `err` variable as opposed to using fresh names. (Ditto elsewhere.)